pkg/machine/sockets/sockets_windows.go

```diff
@@ -0,0 +1,23 @@
+package sockets
+
+import (
+	"fmt"
+	"os"
+	"time"
+
+	"github.com/sirupsen/logrus"
+)
+
+func WaitForSocketWithBackoffs(maxBackoffs int, backoff time.Duration, socketPath string, name string) error {
+	backoffWait := backoff
+	logrus.Debugf("checking that %q is ready", name)
+	for i := 0; i < maxBackoffs; i++ {
+		_, err := os.Stat(socketPath)
```

Comment:
Why split this out? There is really no big reason to use unix.Access() over os.Stat() here so I rather have one function for everything. The duplication is always incredibly hard to work with.

Reply:
I can use os.Stat version for all platforms. I didn't want to change the original implementation, my assumption was that the unix.Access with syscall inside was an intentional optimization (I know this is not a hot path, so, it should not be as critical). I also saw that in some other projects there was a need to use `os.Lstat` for checking socket on Windows https://github.com/lima-vm/lima/blob/e9105bd92211bc25029d2c4b87dff404e043f65e/pkg/httpclientutil/httpclientutil_windows.go#L14, but I could not reproduce the error on recent Windows builds. I can't say if this will have to be addressed additionally or not in the future.

If using `os.Stat` for all platforms will be accepted I will change into unified version.

Reply:
im ok with where this is going ... the timing is bad, but generally this is the direction we were headed.  I would prefer paths remain typed as vmfiles.

in the future, i want a vmsocket type, so that we can check socket len(apple) and do various forms of open, listen, etc. but i had envisioned more of an interface type than by OS.  

lets review this with scrutiny given we are nearing a release and if it looks like any form of improvement, then im ok to merge.

Reply:
>  I would prefer paths remain typed as vmfiles.

`sockets.go` was using VMFile only as an output parameter, I believed this as a design choice and that was the reason to change to plain paths. I can return VMFile, if this is desired.

> in the future, i want a vmsocket type

Then if this or similar change would be introduced it will be easier to find everything what is desired to be changed in one place (or close to each other at least).